### PR TITLE
*: make MVMap.Get() be able to reuse the value buffer pre-allocated

### PR DESCRIPTION
**Pull request by zz-jason**
old result:
```sh
$go test -bench BenchmarkMVMapGet -test.benchmem
goos: darwin
goarch: amd64
pkg: github.com/pingcap/tidb/util/mvmap
BenchmarkMVMapGet-8      5000000               307 ns/op              32 B/op          1 allocs/op
PASS
ok      github.com/pingcap/tidb/util/mvmap      3.996s
```

new result:
```sh
$go test -bench BenchmarkMVMapGet -test.benchmem
goos: darwin
goarch: amd64
pkg: github.com/pingcap/tidb/util/mvmap
BenchmarkMVMapGet-8     10000000               159 ns/op               0 B/op          0 allocs/op
PASS
ok      github.com/pingcap/tidb/util/mvmap      5.300s
```

### CODE REVIEW

**Comment by shenli** [comment]
Any benchmark result?

**Comment by zz-jason** [comment]
@shenli updated in description

File: util/mvmap/mvmap.go [hunk 1]
```diff
@@ -151,8 +151,8 @@ func (m *MVMap) Put(key, value []byte) {
 }
 
 // Get gets the values of the key.
```
Comment (coocood) [inline util/mvmap/mvmap.go]: Please update the comment.

**Comment by coocood** [comment]
@zz-jason 
conflict

**Comment by zz-jason** [comment]
@coocood @shenli PTAL

**Comment by coocood** [comment]
LGTM

**Comment by zz-jason** [comment]
@XuHuaiyu PTAL

File: util/mvmap/mvmap.go [hunk 2]
```diff
@@ -150,9 +150,8 @@ func (m *MVMap) Put(key, value []byte) {
 	m.length++
 }
 
-// Get gets the values of the key.
-func (m *MVMap) Get(key []byte) [][]byte {
-	var values [][]byte
+// Get gets the values of the "key" and append them to "values".
```
Comment (zimulala) [inline util/mvmap/mvmap.go]: s/append/appends

**Comment by zimulala** [comment]
Rest LGTM

**Pull request merged**